### ISSUE TITLE
Recommendation for controller

### ISSUE THREAD

**Issue by FrostKnight** [fgsfdsfgs/perfect_dark#565]
Can you or anyone here recommend a usb controller or a usb joypad controller that would work well? Thanks in advance.

2x is preferred. Also a link to the website and item from which  I can buy it from.

Such as like amazon, ebay, etc... 

All buttons must work or if its an adapter, it must fully work.

Currently my raphnet v3 n64 one doesn't work for the Z, L or R buttons.

**Comment by cylonicboom** [comment]
My 8bitdo controllers and PS5 controller work especially well for modding: I can leave them plugged in for weeks and they never disconnect.

I'm using an Ubuntu derived Linux distro for this. I can't get controllers to stably connect to my MacBook. I haven't tried windows.

**Comment by fgsfdsfgs** [comment]
I have seen other people configure raphnet controllers to work with the port. There seems to be like 200 variations of them in the wild though.

**Comment by LonelySpaceDetective** [comment]
Really depends on your tastes, though anything picked up by SDL (which is a huge variety of controllers) and of course has enough buttons and at least one stick should work fine.
I personally just stick to KB+M, though I can confirm from some experimentation that my DualSense works fine on Windows.

I notice you were using a Raphnet adapter so presumably you were using a real N64 controller, maybe you could try one of the wireless repros Nintendo made for the Switch instead? They actually work on PC as well.

**Comment by FrostKnight** [comment]
> I have seen other people configure raphnet controllers to work with the port. There seems to be like 200 variations of them in the wild though.

Yeah... its weird.  I have the dual n64 version 3 one.  The regular dual n64 v3 one.

**Comment by FrostKnight** [comment]
> Really depends on your tastes, though anything picked up by SDL (which is a huge variety of controllers) and of course has enough buttons and at least one stick should work fine. I personally just stick to KB+M, though I can confirm from some experimentation that my DualSense works fine on Windows.
> 
> I notice you were using a Raphnet adapter so presumably you were using a real N64 controller, maybe you could try one of the wireless repros Nintendo made for the Switch instead? They actually work on PC as well.

What about for linux? and which one specifically would you recommend. Can you show me one that is good on amazon or ebay or something? I kind of need one that works...

Like would these work?:

https://www.amazon.com/Wireless-Controller-Receiver-Compatible-Nintendo-64/dp/B0D7H458RM/ref=sr_1_5?dib=eyJ2IjoiMSJ9.gW3uO49bhMrGv0yCKe05ZkrVlpB64SQNtHR3V-vJxaUidXSSEOjJ9r2kBKfIAgqzHLp1XFv028BRxYsalNVIyp_l4mAQzWAMCRBPumIu7FiTvn3nM1NPfvYH_zhwjiF3uQupiy2P8gAvbESv34-DVHXou6_XciVvI_mF4ZMPRUEvDSVN7ufe3Tnqfmdg2q_ayebegfdA7y22wIhtKzvI1sbzdkVC8IchO_XqE5snnZQ.QN6upEP57zCpTF7ShA5-xaMDEXZyxhcYALUXd6lC0T0&dib_tag=se&keywords=n64+wireless+controller&qid=1734944487&sr=8-5

or these?:

https://www.amazon.com/Wireless-Controller-Compatible-Raspberry-Nintendo-64/dp/B0C98MGV5G/ref=sr_1_1_sspa?dib=eyJ2IjoiMSJ9.gW3uO49bhMrGv0yCKe05ZkrVlpB64SQNtHR3V-vJxaUidXSSEOjJ9r2kBKfIAgqzHLp1XFv028BRxYsalNVIyp_l4mAQzWAMCRBPumIu7FiTvn3nM1NPfvYH_zhwjiF3uQupiy2P8gAvbESv34-DVHXou6_XciVvI_mF4ZMPRUEvDSVN7ufe3Tnqfmdg2q_ayebegfdA7y22wIhtKzvI1sbzdkVC8IchO_XqE5snnZQ.QN6upEP57zCpTF7ShA5-xaMDEXZyxhcYALUXd6lC0T0&dib_tag=se&keywords=n64+wireless+controller&qid=1734944654&sr=8-1-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&psc=1

Basically give me some ideas of specifics.

**Comment by FrostKnight** [comment]
Actually, do you know of any adapters/controllers that will work that have the control stick in the traditional spot?

like here: https://www.amazon.com/Classic-Controller-iNNEXT-Joystick-Raspberry/dp/B07449HLKJ/ref=sr_1_3?crid=2YKSMGT9IJY8T&dib=eyJ2IjoiMSJ9.sQ-ZDIxSgv2F49ryHqoAdeOQC2ZYnDPNWhqtMDLf0L7omTrcWyczQOJtLgr_VO2oel_4zhf06lDwO2vDEJlCP0mOD5H0r6fD_R5vy1ebzJzDSaNJ0Bn7s6rOQH_zOu3smoez1aP5Xex2fVcw92exm24nTpy3DR8N40PuC1zPTYM3IbJhseORW6g9GH4c4qVejmlLfgp7uQnBIkHlXSBFceYVx6kwXfjfsh-WXzBQRhE.UMuHqDjjlvgCHvdtbtALkbf7dCf7s5eB6U6M4YZuk8Y&dib_tag=se&keywords=adaptoid+n64+controller&qid=1734945972&s=videogames&sprefix=adapton64+controller%2Cvideogames%2C323&sr=1-3

And not here: 

https://www.amazon.com/sspa/click?ie=UTF8&spc=MTo4NDQ1MjY5OTU4ODQyMzExOjE3MzQ5NDU5NzI6c3BfYXRmOjIwMDAzODg1Nzc4Njk3MTo6MDo6&url=%2FWarrior-Controller-Original-Nintendo-Console%2Fdp%2FB08GJMGRXV%2Fref%3Dsr_1_1_sspa%3Fcrid%3D2YKSMGT9IJY8T%26dib%3DeyJ2IjoiMSJ9.sQ-ZDIxSgv2F49ryHqoAdeOQC2ZYnDPNWhqtMDLf0L7omTrcWyczQOJtLgr_VO2oel_4zhf06lDwO2vDEJlCP0mOD5H0r6fD_R5vy1ebzJzDSaNJ0Bn7s6rOQH_zOu3smoez1aP5Xex2fVcw92exm24nTpy3DR8N40PuC1zPTYM3IbJhseORW6g9GH4c4qVejmlLfgp7uQnBIkHlXSBFceYVx6kwXfjfsh-WXzBQRhE.UMuHqDjjlvgCHvdtbtALkbf7dCf7s5eB6U6M4YZuk8Y%26dib_tag%3Dse%26keywords%3Dadaptoid%2Bn64%2Bcontroller%26qid%3D1734945972%26s%3Dvideogames%26sprefix%3Dadapton64%2Bcontroller%252Cvideogames%252C323%26sr%3D1-1-spons%26sp_csd%3Dd2lkZ2V0TmFtZT1zcF9hdGY%26psc%3D1

**Comment by LonelySpaceDetective** [comment]
> What about for linux?

Yeah sorry, I'm not a Linux guy, though I think the DualSense works on Linux and SDL picks it up.

As for the specific links, well I mean, sorry but I can't really personally confirm. They _probably_ will work or can be made to work, but I'm not the person to ask.
All I can say is that none of them are the official repros by Nintendo I referred to.

> and which one specifically would you recommend

Assuming you meant a controller in general, well, if I were to play an FPS with a gamepad I'd probably pull out my DualSense as it feels good in the hand and I can do gyro aiming, or my Xbox Core Controller if I want something that Just Works and I don't need to do any fiddling with, but you sound like you want specifically an N64 or N64-like controller, which a DualSense or Xbox controller obviously isn't, and I don't really have any experience on that front.

**Comment by FrostKnight** [comment]
Yes, does any adapter you suggest supported with this type of controller? 
![71phBtmUt-L __AC_SX300_SY300_QL70_ML2_](https://github.com/user-attachments/assets/8e1b0c81-9fd3-4acd-8911-05bdb3b2e176)
This goes for everyone who responds btw.

**Comment by FrostKnight** [comment]
> I have seen other people configure raphnet controllers to work with the port. There seems to be like 200 variations of them in the wild though.

Do you know any n64 adapters that work with the controller picture I attached? just now above this new comment I am sending you?

For linux preferably. Although windows configs might also work, I dont know.
